### ISSUE TITLE
read arrow fails with unknown type ArrowType$LargeUtf8

### ISSUE THREAD

**Issue by behrica** [techascent/tech.ml.dataset#147]
A big arrow file fails with:

```
Execution error at tech.v3.libs.arrow.in-place/read-schema$fn (in_place.clj:93). 
Failed to datafy datatype class org.apache.arrow.vector.types.pojo.ArrowType$LargeUtf8
```

**Comment by behrica** [comment]
I wrote he smae file, but chping of moste of teh rows.
Then it works weel.

So it seems that type "ArrowType$LargeUtf8" is used for large string columns

**Comment by behrica** [comment]
The R package for arrow seems already well working.

So I see here a good opportunity to use it as a new "exchange format" between R and Clojure code.

I was so often bitten badly in the incompatibilities between R and other tools regarding CSV (especially if working with text columns) that I am happy to invest some time to make arrow work well in  T.d.m.

You have done already a very good job, so I can at least supporting you with testing it

**Comment by cnuernber** [comment]
Could I get access to the file?  This should be pretty straight forward to support.

**Comment by behrica** [comment]
I cannot upload easely a 4 G file to the internet over ADLS line.
It should happen wit any large enough arrow file with a text column and a few million rows.
I generated one from R, with R package arrow

**Comment by cnuernber** [comment]
Ok, that makes sense.  I will work with it a bit and we can see :-)

**Comment by cnuernber** [comment]
What about attaching the script you used to generate the table?  Then I can just run it from the command line.  The data can be totally made up.

**Comment by cnuernber** [comment]
In fact, what about a script specifically for this problem and then a fuzz script that generates a range of ideally odd or breaking datatypes.

**Comment by behrica** [comment]
R code to create arrow file with 10 000 000 rows
6G on disk

```R
library(arrow)
df=data_frame(texts=rep("A character vector containing abbreviations for the character strings in its first argument. Duplicates in the original names.arg will be given identical abbreviations. If any non-duplicated elements have the same minlength abbreviations then, if method = both.sides the basic internal abbreviate() algorithm is applied to the characterwise reversed strings; if there are still duplicated abbreviations and if strict = FALSE as by default, minlength is incremented by one and new abbreviations are found for those elements only. This process is repeated until all unique elements of names.arg have unique abbreviations.",10000000)
)
arrow::write_ipc_stream(df,"10m.arrow")
```

```clojure
(arrow/read-stream-dataset-inplace "10m.arrow")
-->
Execution error at tech.v3.libs.arrow.in-place/read-schema$fn (in_place.clj:93).
Failed to datafy datatype class org.apache.arrow.vector.types.pojo.ArrowType$LargeUtf8
```

Having it smaller, like 3M rows, works

**Comment by cnuernber** [comment]
Fixed in `5.00-alpha-11`
